setup.py

```diff
@@ -32,7 +32,7 @@
         'pyasn1>=0.0.13',
```

Comment:
I apparently can't comment on the rest of the file, but you can remove the distutils try: except: from the top.  New pip won't manage distutils installs and setuptools is pretty universal